### PR TITLE
Feature/#848 write meaningful tests

### PR DESCRIPTION
**Pull request by gobbletsit**
https://app.productive.io/1-infinum/projects/2590/tasks/task/3298988?board=85892&filter=MjQ1OTU5

This PR contains changes made for covering the existing functionality with tests as much as possible. Plenty of errors were detected with newly written tests so new tests were added for all those newly encountered issues.

Done:

- Cover type adapters with tests as much as possible
- Fix newly detected issues
- Cover newly detected cases with test if possible

### CODE REVIEW

File: core/src/main/java/com/infinum/jsonapix/core/discriminators/BaseJsonApiDiscriminator.kt [hunk 1]
```diff
@@ -92,10 +93,12 @@ abstract class BaseJsonApiDiscriminator(
...
             val set = relationshipEntry.value.jsonObject.entries.toMutableSet()
-            getLinksObject(relationshipEntry.value)?.let { linksSafe ->
-                val newLinks = relationshipsLinksDiscriminator.inject(linksSafe)
-                set.removeAll { it.key == JsonApiConstants.Keys.LINKS }
-                set.add(getJsonObjectEntry(JsonApiConstants.Keys.LINKS, newLinks))
+            getLinksObject(relationshipEntry.value)?.let { linksObject ->
+                if (linksObject !is JsonNull) {
+                    val newLinks = relationshipsLinksDiscriminator.inject(linksObject)
+                    set.removeAll { it.key == JsonApiConstants.Keys.LINKS }
+                    set.add(getJsonObjectEntry(JsonApiConstants.Keys.LINKS, newLinks))
```
Comment (stjepanbanek) [inline core/src/main/java/com/infinum/jsonapix/core/discriminators/BaseJsonApiDiscriminator.kt:100]: Not sure about this, but do we need to do the same for meta?

**Review by stjepanbanek — APPROVED** [review]
Looks good! 💪

File: core/src/main/java/com/infinum/jsonapix/core/discriminators/BaseJsonApiDiscriminator.kt (hunk 1 above)
Comment (gobbletsit) [reply core/src/main/java/com/infinum/jsonapix/core/discriminators/BaseJsonApiDiscriminator.kt:100]: Yes we do, missed that, will add the handling. Good catch 👍

**Review by stjepanbanek — APPROVED** [review]
Good work!

**Pull request merged**